pkg/dbal/DbalProducer.php

```diff
@@ -74,15 +75,12 @@ public function send(PsrDestination $destination, PsrMessage $message)
             ));
         }
 
-        $sql = 'SELECT '.$this->context->getDbalConnection()->getDatabasePlatform()->getGuidExpression();
-        $uuid = $this->context->getDbalConnection()->query($sql)->fetchColumn(0);
-
-        if (empty($uuid)) {
-            throw new \LogicException('The generated uuid is empty');
-        }
+        $hasNativeGuid = $this->context->getDbalConnection()->getDatabasePlatform()->hasNativeGuidType();
+        $uuid = Uuid::uuid4();
```

Comment:
Why not `Uuid::uuid1()`? It is more efficient for the database to index than a random v4 UUID.

Reply:
Honestly, I dont know I've been using uuid4 everywhere. I'll look into it. 

Thanks.

Reply:
good write up on this  https://www.sohamkamani.com/blog/2016/10/05/uuid1-vs-uuid4/

Reply:
@makasim check the link from Percona linked in my comment.

Reply:
@dkarlovi yeah, I saw it. Unfortunately I am not sure we can adopt it since it is MySQL\InnoDB  specific stuff. We have to support or DBAL transport hence the solution should be flexible enough to work on all of those platforms

Reply:
No, I meant the influence of random on performance. v1 vs v4 should be considered here IMO.

Reply:
ah, I see